drivers/timer/cortex_m_systick.c

```diff
@@ -152,25 +186,27 @@ static uint32_t elapsed(void)
 		val2 = last_load;
 	}
 
-	if ((ctrl & SysTick_CTRL_COUNTFLAG_Msk)
-	    || (val1 < val2)) {
+	if ((ctrl & SYST_CSR_COUNTFLAG) || (val1 < val2)) {
 		overflow_cyc += last_load;
 
 		/* We know there was a wrap, but we might not have
-		 * seen it in CTRL, so clear it. */
-		(void)SysTick->CTRL;
+		 * seen it in SYST_CSR, so clear it.
+		 */
+		sys_read32(SYST_CSR);
 	}
 
 	return (last_load - val2) + overflow_cyc;
 }
 
-/* Callout out of platform assembly, not hooked via IRQ_CONNECT... */
-void sys_clock_isr(void *arg)
+/* sys_clock_isr is calling directly from the platform's vectors table.
+ * Also ISR_DIRECT_DECLARE() makes it ready to use and doesn't require any
+ * additional hooks via IRQ_CONNECT/IRQ_DIRECT_CONNECT
+ */
+ISR_DIRECT_DECLARE(sys_clock_isr)
 {
-	ARG_UNUSED(arg);
-	uint32_t dcycles;
 	uint32_t dticks;
 
+	k_spinlock_key_t key = k_spin_lock(&lock);
```

Comment:
Inconsistent style.

Also, I am not sure how advisable it is to spinlock in this ISR.
AFAIU, there shoul not be use of these API at higher priority than this ISR.

Reply:
My fault in comment (the code literally is more then 2 years old), actually it's not for higher level ISR, but mainly for the latest multi-cores Cortexes. IMO it's better to slow down by noop interrupt disable, then debugging race condition after.